app/assets/javascripts/discourse/app/components/topic-timeline/scroll-area.js

```diff
@@ -0,0 +1,231 @@
+import GlimmerComponent from "discourse/components/glimmer";
+import { bind } from "discourse-common/utils/decorators";
+import { tracked } from "@glimmer/tracking";
+import discourseLater from "discourse-common/lib/later";
+import { action } from "@ember/object";
+
+export const SCROLLER_HEIGHT = 50;
+const MIN_SCROLLAREA_HEIGHT = 170;
+const MAX_SCROLLAREA_HEIGHT = 300;
+const LAST_READ_HEIGHT = 20;
+
+export default class TopicTimelineScrollArea extends GlimmerComponent {
+  @tracked showButton = false;
+  @tracked scrollPosition;
+  @tracked current;
+  @tracked percentage = this._percentFor(
+    this.args.topic,
+    this.args.enteredIndex + 1
+  );
```

Comment:
This will need to be a getter to make sure it's re-evalated when the args change